deps/media-playback/media-playback/media.c

```diff
@@ -226,17 +225,68 @@ static bool mp_media_init_scaling(mp_media_t *m)
 	sws_setColorspaceDetails(m->swscale, coeff, range, coeff, range, 0,
 				 FIXED_1_0, FIXED_1_0);
 
-	int ret = av_image_alloc(m->scale_pic, m->scale_linesizes,
-				 m->v.decoder->width, m->v.decoder->height,
-				 m->scale_format, 32);
+	int ret = av_image_alloc(m->scale_pic, m->scale_linesizes, frame->width,
+				 frame->height, dst_fmt, 32);
 	if (ret < 0) {
 		blog(LOG_WARNING, "MP: Failed to create scale pic data");
 		return false;
 	}
 
+	m->scale_format = dst_fmt;
+	m->scale_range = range;
+	m->scale_space = space;
+	m->scale_src_format = frame->format;
+	m->scale_src_width = frame->width;
+	m->scale_src_height = frame->height;
+
 	return true;
 }
 
+static bool mp_media_check_scaling(mp_media_t *m)
+{
+	bool new_scaling_needed = false;
+	enum AVPixelFormat scale_fmt = closest_format(m->v.frame->format);
+	int space = get_sws_colorspace(m->v.frame->colorspace);
+	int range = get_sws_range(m->v.frame->color_range);
+
+	if (scale_fmt == m->v.frame->format) {
+		sws_freeContext(m->swscale);
```

Comment:
Should be noted that this only works (see also `mp_media_free`) because `sws_freeContext` and `av_freep` accept `NULL` pointers (which would be the case if a prior frame required no scaling as well).